### PR TITLE
[ECO-4841] Add README section to explain "Connection limit exceeded" error and solutions

### PR DESCRIPTION
**Pull request by VeskeR**
Resolves [ECO-4841](https://ably.atlassian.net/browse/ECO-4841)

This is based on the "max connections errors" research done in [ECO-4841](https://ably.atlassian.net/browse/ECO-4841) (see comments), "[Ably-js max connections issue overview and possible solutions](https://ably.atlassian.net/wiki/spaces/SDK/pages/3381854247/Ably-js+max+connections+issue+overview+and+possible+solutions)", and solutions found in https://github.com/ably-labs/react-hooks/issues/8#issuecomment-1697410212 and https://github.com/ably/ably-js/issues/1742#issuecomment-2066179610

Related: https://github.com/ably/ably-js/issues/1779, https://github.com/ably/ably-js/issues/1742

[ECO-4841]: https://ably.atlassian.net/browse/ECO-4841?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ECO-4841]: https://ably.atlassian.net/browse/ECO-4841?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

- **New Features**
	- Added guidance for managing the "Connection limit exceeded" error in the README.
	- Introduced new hooks: `useConnectionStateListener`, `useChannelStateListener`, and `useAbly`.
	- New parameter `skip` added to `useChannel`, `usePresence`, and `usePresenceListener` hooks for better control over attachment behavior.
  
- **Documentation**
	- Enhanced clarity and expanded usage examples for `ably-js React Hooks`.
	- Updated error handling section with detailed management strategies for connection and channel errors.
	- Added a section addressing common warnings when using the library with Next.js.

### CODE REVIEW

**Comment by VeskeR** [comment]
@coderabbitai pause

File: README.md [hunk 1]
```diff
@@ -641,6 +641,26 @@ The issue is coming from the fact that when using App Router specifically depend
 Using `serverComponentsExternalPackages` opt-outs from using Next.js bundling for specific packages and uses native Node.js `require` instead.
 This is a common problem in App Router for a number of packages (for example, see next.js issue [vercel/next.js#52876](https://github.com/vercel/next.js/issues/52876)), and using `serverComponentsExternalPackages` is the recommended approach here.
 
+#### "Connection limit exceeded" error during development
```
Comment (owenpearson) [inline README.md:644]:
```suggestion
#### "Connection limit exceeded" error during development with React
```

**Review by owenpearson — APPROVED** [review]
lgtm, one minor suggestion

File: README.md (hunk 1 above)
Comment (owenpearson) [reply README.md:644]: ah yep, good point 👍

**Review by owenpearson — APPROVED** [review]

**Review by ttypic — APPROVED** [review]
LGTM

**Pull request merged**